### PR TITLE
add node ConditioningTimestepInterpolate

### PR DESCRIPTION
**Pull request by silveroxides**
Adds node for interpolating between two conditionings at a set interval.
Simplifies something that would otherwise require over 30 nodes ifor generating an image interpolating every step for 20 steps with currently available Core nodes. 
It also mimics prompt editing from A1111 webui that did similar thing.

### CODE REVIEW

File: nodes.py [hunk 1]
```diff
@@ -264,6 +264,52 @@ def set_range(self, conditioning, start, end):
...
+        return {"required": {"conditioning_1": ("CONDITIONING", ),
+                             "conditioning_2": ("CONDITIONING", ),
+                             "interval_range": ("FLOAT", {"default": 0.25, "min": 0.001, "max": 0.5, "step": 0.001})
+                             }}
+    RETURN_TYPES = ("CONDITIONING",)
+    FUNCTION = "interpolate_conds"
+
+    CATEGORY = "advanced/conditioning"
+
+
```
Comment (mcmonkey4eva) [inline nodes.py]: (minor nit) stray lines here - try to match the format of surrounding code

File: nodes.py [hunk 2]
```diff
@@ -264,6 +264,52 @@ def set_range(self, conditioning, start, end):
...
+            if i % 2 == 0:
+                conditioning_1_intervals.append(node_helpers.conditioning_set_values(conditioning_1, {"start_percent": start, "end_percent": end}))
+            else:
+                conditioning_2_intervals.append(node_helpers.conditioning_set_values(conditioning_2, {"start_percent": start, "end_percent": end}))
+
+        # Combine intervals for conditioning_1 and conditioning_2
+        conditioning_1_intervals_combined = conditioning_1_intervals[0]
+        for i in range(1, len(conditioning_1_intervals)):
+            conditioning_1_intervals_combined = (conditioning_1_intervals_combined + conditioning_1_intervals[i])
```
Comment (mcmonkey4eva) [inline nodes.py:300]: what's going on here? this seems redundant

Comment (silveroxides) [reply nodes.py:300]: The iteration through all the intervals or the combine part?

Comment (mcmonkey4eva) [reply nodes.py:300]: the combine part - like you already have a loop at the top to build a list, why are you unbuilding and rebuilding the same list?

**Comment by asagi4** [comment]
I'm a bit confused about this node. Where's the interpolation happening? Did you mean interleaving, or am I just completely misreading the code?

File: nodes.py (hunk 2 above)
Comment (silveroxides) [reply nodes.py:300]: I didn't at first but conditioning did not accept a list. I am only doing the same operation as the ConfitioningCombine is doing.

**Comment by silveroxides** [comment]
> I'm a bit confused about this node. Where's the interpolation happening? Did you mean interleaving, or am I just completely misreading the code?

How would you approach setting up interpolation? It definitely isn't interleaving

**Comment by asagi4** [comment]
@silveroxides then I don't understand what the code is doing, because it doesn't look like interpolation to me, but alternating between cond1 and cond2 definitely looks like interleaving to me.

**Comment by asagi4** [comment]
As for interpolation, I'd expect an interpolating node to result in a set of conditionings that gradually move from cond1 towards cond2, which doesn't seem to be happening here.

**Comment by silveroxides** [comment]
@asagi4 you're right. It was me who had gotten them mixed up due to them being semantically similar. although I would not call a gradual transition from one cond to another for interpolation  either. I doubt this will be merged anyway. It was more a call to attention regarding the absurd amount of nodes needed to perform same operation as [red|green] does in a1111.

**Comment by asagi4** [comment]
@silveroxides You could use my prompt control nodes if that functionality is what you want; excessive noodle soup is the reason I wrote them.

An alternating node could still be useful in core too, though.

**Comment by silveroxides** [comment]
@asagi4 You havent considered making a PR for that functionality to be part of core? Seriously the amount of users that miss that functionality yet want/need the memory effiecency ComfyUI offers are many. There are bunch of other basic node editor node types that is completely missing too. Like anyone who have used node editors in the past would expect things like basic logic to be included in core. Right now, in order to be able to do a full range of logic operations you need several custom node packs because each one only serves the logic node it needs xD

**Comment by asagi4** [comment]
@silveroxides The functionality is a bit too complex to just add into core as is, though I have thought about proposing some changes to core so that I could get rid of sampler monkey patching. Just never got around to implementing that.

**Pull request closed without merging**